plugins/ksp/src/org/jetbrains/kotlin/ksp/processor/ResolveJavaTypeProcessor.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.ksp.processor
+
+import org.jetbrains.kotlin.ksp.processing.Resolver
+import org.jetbrains.kotlin.ksp.symbol.KSClassDeclaration
+import org.jetbrains.kotlin.ksp.symbol.KSFunctionDeclaration
+import org.jetbrains.kotlin.ksp.symbol.KSNode
+import org.jetbrains.kotlin.ksp.symbol.KSTypeReference
+import org.jetbrains.kotlin.ksp.symbol.impl.java.KSClassDeclarationJavaImpl
+import org.jetbrains.kotlin.ksp.symbol.impl.java.KSTypeReferenceJavaImpl
+import org.jetbrains.kotlin.ksp.visitor.KSTopDownVisitor
+
+class ResolveJavaTypeProcessor : AbstractTestProcessor() {
+    val results = mutableListOf<String>()
+    val visitor = ResolveJavaTypeVisitor()
+
+    override fun toResult(): List<String> {
+        return results
+    }
+
+    override fun process(resolver: Resolver) {
+        val annotated = resolver.getSymbolsWithAnnotation("Test")
+        (annotated.single() as KSClassDeclaration).superTypes.map { it.resolve()?.declaration }.filterIsInstance<KSClassDeclarationJavaImpl>().map { it.accept(visitor, Unit) }
```

Comment:
Can we use `KSClassDeclaration` rather than `KSClassDeclarationJavaImpl`?

Reply:
This is for the test to quickly filter out non-Java symbols since we don't care non-Java symbols in this test.

Reply:
I guess being defined in a java source file already guaranteed it? Nevertheless, it should be better to use only public API since this is an end-to-end test.

Reply:
Will address this once #39 is merged.